cmd/kubeadm/app/apis/kubeadm/v1alpha1/defaults.go

```diff
@@ -89,6 +96,22 @@ func SetDefaults_MasterConfiguration(obj *MasterConfiguration) {
 	if obj.Etcd.DataDir == "" {
 		obj.Etcd.DataDir = DefaultEtcdDataDir
 	}
+
+	if obj.Etcd.SelfHosted.ClusterSize == 0 {
```

Comment:
move all this to a `SetDefaults_Etcd_SelfHosted` and call that from here for cleanliness?